### PR TITLE
ci: Use `bazel.yml` for Docker publishing

### PR DESCRIPTION
**Pull request by phlax**
Cleanups:

- use bazel.yml for Docker publishing
- simplify publishing scripts

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: .azure-pipelines/stage/publish.yml [hunk 1]
```diff
@@ -204,8 +231,101 @@ jobs:
           set -e
           rm -rf $(Build.StagingDirectory)/.gnupg
 
+- job: signed_bins
```
Comment (phlax) [inline .azure-pipelines/stage/publish.yml:234]:
```suggestion
- job: signed_release
```

File: .azure-pipelines/stage/publish.yml [hunk 2]
```diff
@@ -204,8 +231,101 @@ jobs:
...
+  condition: |
+    and(not(canceled()),
+        eq(${{ parameters.publishDocs }}, 'true'))
+  pool:
+    vmImage: "ubuntu-20.04"
+  steps:
+  - template: ../bazel.yml
+    parameters:
+      ciTarget: docs
+      cacheTestResults: ${{ parameters.cacheTestResults }}
```
Comment (phlax) [inline .azure-pipelines/stage/publish.yml:282]: there was a reason i added this altho struggling to rem what it was - i think we will want this but im not sure its immediately necessary

File: distribution/BUILD [hunk 3]
```diff
@@ -73,3 +75,49 @@ sh_binary(
...
+)
+
+genrule(
+    name = "signed",
+    outs = ["release.signed.tar.zst"],
+    cmd = """
+    # Sign the packages
+    $(location //tools/distribution:sign) \
+        "deb.x64:$(location :x64-packages)" \
+        "deb.arm64:$(location :arm64-packages)" \
```
Comment (phlax) [inline distribution/BUILD:106]: currently these packages are added without mapping the deb/changes files into the `bin/` folder which is what we want for release

File: distribution/binary/BUILD [hunk 4]
```diff
@@ -43,3 +43,16 @@ pkg_tar(
     compressor_args = "-T0",
     extension = "tar.zst",
 )
+
+label_flag(
+    name = "release-bundle",
+    build_setting_default = ":release",
+)
+
+pkg_tar(
```
Comment (phlax) [inline distribution/binary/BUILD:52]: i think this is unused

File: .azure-pipelines/stage/publish.yml [hunk 5]
```diff
@@ -93,70 +102,91 @@ jobs:
...
+      cacheKeyDocker: "ci/Dockerfile-envoy | VERSION.txt| $(cacheKeyBazelFiles)"
+      cacheKeyDockerName: publish_docker
+      cacheKeyDockerTmpDir: /var/azpcache
+      cacheKeyDockerPath: ""
+      cacheKeyDockerVersion: "$(cacheKeyDockerBuild)"
+      env:
+        GCS_ARTIFACT_BUCKET: ${{ parameters.bucketGCP }}
+      stepsPre:
+      - bash: |
+          echo "disk space at beginning of build:"
```
Comment (phlax) [inline .azure-pipelines/stage/publish.yml:127]:
```suggestion
          echo "disk space at beginning of Docker build:"
```

**Review by wbpcode — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**